migrations/manual/educandu-2021-12-13-01-make-cdn-urls-generic.js

```diff
@@ -0,0 +1,55 @@
+async function updateAll(collection, query, updateFn) {
+  const cursor = collection.find(query);
+
+  /* eslint-disable-next-line no-await-in-loop */
+  while (await cursor.hasNext()) {
+    /* eslint-disable-next-line no-await-in-loop */
+    const doc = await cursor.next();
+    /* eslint-disable-next-line no-await-in-loop */
+    const updatedDoc = await updateFn(doc);
+    if (updatedDoc) {
+      /* eslint-disable-next-line no-await-in-loop */
+      await collection.replaceOne({ _id: doc._id }, updatedDoc);
+    }
+  }
+}
+
+export function processSection(section) {
+  let updateCount = 0;
+  if (section.type === 'markdown' && section.content) {
+    const regexp = /\]\((https:\/\/((staging\.)?cdn\.elmu\.online)|(https:\/\/cdn\.((integration|staging)\.)?openmusic\.academy))\//g;
```

Comment:
After seeing this, I'm thinking that some migrations should be ran in elmu/openmusic separately and not be a part of educandu. It's convenient that we have them here, but educandu as the library should not know the specifics of the systems using it.

Reply:
The reason I prefixed the migrations with `educandu` was, that in case we have later 3rd party migrations (e.g. from plugins), we can distinguish them in the migrations collections. But for now we don't have the necessary system in place to easily mix them. I would suggest anyway to reset to one single migration again after each milestone, as long as we are in control of all implementing systems.

Reply:
That's true..
We'll anyway have to revisit how we run migration scripts, not only from this perspective 😬